### PR TITLE
Nano: Add a multi-stage pipeline

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add a two stage pipeline, transfer data between stages using `multiprocessing.Pipe`. 

### 2. User API changes
like this:
```python
from bigdl.nano.pytorch.inference.pipeline import Pipeline

pipeline = Pipeline([
    ("preprocess", preprocess, {}),
    ("inference", model, {}),
])
input_paths = [
    "/root/yishuo/test.jpg"
    for i in range(100)
]
outputs = pipeline.run(input_paths)
```

**A complete example:**
```python
import time

import torch
from torchvision import transforms
from torchvision.models import resnet50

from PIL import Image

from bigdl.nano.pytorch.inference.pipeline import Pipeline


transform = transforms.Compose([
    transforms.Resize(256),
    transforms.ColorJitter(),
    transforms.Resize(128),
    transforms.ToTensor(),
])


def preprocess(img_path: str):
    img = Image.open(img_path)
    img = transform(img)
    batch = torch.stack([img for i in range(8)], 0)
    return batch


def test_pipeline(model):
    pipeline = Pipeline([
        ("preprocess", preprocess, {}),
        ("inference", model, {}),
    ])

    input_paths = [
        "/root/yishuo/test.jpg"
        for i in range(100)
    ]
    outputs = pipeline.run(input_paths)


def test_normal(model):
    input_paths = [
        "/root/yishuo/test.jpg"
        for i in range(100)
    ]
    with torch.inference_mode(True):
        for p in input_paths:
            batch = preprocess(p)
            _ = model(batch)

# warmup
model = resnet50()
with torch.inference_mode(True):
    for i in range(5):
        inputs = preprocess("/root/yishuo/test.jpg")
        _ = model(inputs)

st = time.time()
inputs = preprocess("/root/yishuo/test.jpg")
print('preprocess time:', time.time() - st)

st = time.time()
with torch.inference_mode(True):
    _ = model(inputs)
print('inference time:', time.time() - st)

st = time.time()
test_pipeline(model)
print('total pipeline time:', time.time() - st)

st = time.time()
test_normal(model)
print('total baseline time:', time.time() - st)
```

The output maybe:
```
preprocess time: 0.0673820972442627
inference time: 0.06682658195495605
total pipeline time: 8.78221607208252
total baseline time: 13.228902101516724
```

### 4. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: python/nano/src/bigdl/nano/pytorch/inference/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import itertools
+import multiprocessing as mp
+from multiprocessing.connection import Connection
+from typing import List, Tuple, Dict, Callable
+
+from torch import nn
+
+
+class Pipeline():
+    def __init__(self, stages: List[Tuple]):
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/pytorch/inference/pipeline.py:26]: please add some api doc string here and make it shown on api doc page.

File: python/nano/src/bigdl/nano/pytorch/inference/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    def _validate_name(name: str):
+        pass
+
+    @staticmethod
+    def _validate_config(config: Dict):
+        pass
+
+    @staticmethod
+    def _apply_config(config: Dict):
+        pass
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/pytorch/inference/pipeline.py:81]: maybe we could just remove these validate methods for the first push.

Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/pytorch/inference/pipeline.py:81]: please add an example to https://github.com/intel-analytics/BigDL/tree/main/python/nano/tutorial/inference/pytorch and some ut

Comment (MeouSker77) [reply python/nano/src/bigdl/nano/pytorch/inference/pipeline.py]: have added some UT, I'll add an example in next PR

**Review by TheaperDeng — APPROVED** [review]
LGTM, seems to be a good start point for this function.

**Pull request merged**